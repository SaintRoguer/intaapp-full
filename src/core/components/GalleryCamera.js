import {Container, Content, Spinner} from 'native-base';
import React, {Component} from 'react';
import {StyleSheet} from 'react-native';
import {mainThemeColor, galleryCameraButtons} from '../../configuration';
import {
  HomeCard,
  withImagePicker,
  withImageProcessing,
  withImageHandler,
} from '../../shared';
import {ImageModel} from '../../shared/models/ImageModel.js';

class GalleryCamera extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getGalleryCameraButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getGalleryCameraButtons() {
    return galleryCameraButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    try {
      this.setState({loading: true});
      const imageResponse = await this.props.imageHandler.pickImage()(picker);
      this.props.navigation.navigate('Imagen', imageResponse);
    } finally {
      this.setState({loading: false});
    }
  };
}

export default withImageHandler(GalleryCamera);
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
