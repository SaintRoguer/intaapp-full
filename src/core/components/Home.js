import {Container, Content, Spinner} from 'native-base';

import React, {Component} from 'react';
import {StyleSheet, Alert} from 'react-native';
import {mainThemeColor, homeButtons} from '../../configuration';
import {HomeCard, ImagePickerService, ImageProcessor} from '../../shared';

export class Home extends Component {
  picker;
  imageProcessor;

  constructor(props) {
    super(props);
    this.picker = ImagePickerService.getInstance();
    this.imageProcessor = ImageProcessor.getInstance();
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getHomeButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getHomeButtons() {
    return homeButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    this.props.navigation.navigate('GalleryCamera', {});
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // backgroundColor: '#20d2bb',
  },
});
