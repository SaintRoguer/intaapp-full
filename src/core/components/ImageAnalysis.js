//@ts-check
import React, {useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {SaveImage, ImageView, Tabs, ImageNotes} from '../../shared';
import {withImageProcessing} from '../../shared';

function ImageAnalysis({route: {params}, imageProcessor}) {
  const onSave = params.onSave;
  const [originalImage, setOriginalImage] = useState(params.originalImage);
  const [processedImage, setProcessedImage] = useState(params.processedImage);
  const [percentages, setPercentages] = useState(params.percentages);
  const [note, setNote] = useState('');

  const updateOriginalImage = async newUri => {
    let {
      percentageGreen,
      percentageYellow,
      percentageNaked,
      img,
    } = await imageProcessor.processImage('file://' + newUri);
    setPercentages({
      percentageGreen,
      percentageYellow,
      percentageNaked,
    });

    setOriginalImage(prev => {
      const newOriginal = prev;
      newOriginal.setUri('file://' + newUri);
      return newOriginal.clone();
    });
    setProcessedImage(prev => {
      const newProcessed = prev;
      newProcessed.setData(img);
      return newProcessed.clone();
    });
  };

  const saveImage = async () => {
    return await onSave(percentages, originalImage.getSource(), note);
  };

  const getImageView = () => (
    <ImageView
      shouldRotate={params.shouldRotate}
      updateOriginalImage={updateOriginalImage}
      originalImage={originalImage}
      staticOriginal={params.originalImage.clone()}
      processedImage={processedImage}
      percentages={percentages}
    />
  );

  return (
    <View style={styles.container}>
      {onSave ? (
        <Tabs
          FirstScreen={getImageView}
          SecondScreen={() => <ImageNotes value={note} updateNote={setNote} />}
          secondTitle="Notas"
          firstTitle="AnÃ¡lisis"
        />
      ) : (
        getImageView()
      )}
      {onSave ? <SaveImage onSave={saveImage} /> : null}
    </View>
  );
}
export default withImageProcessing(ImageAnalysis);

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
