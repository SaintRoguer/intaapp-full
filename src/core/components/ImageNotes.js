import React from 'react';
import {View} from 'react-native';

export function ImageNotes() {
  return <View />;
}
