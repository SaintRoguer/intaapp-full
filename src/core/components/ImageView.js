import React, {Component} from 'react';
import {StyleSheet, View} from 'react-native';
import SwitchSelector from 'react-native-switch-selector';
import {options, switchConfig} from '../../configuration';
import {
  ImageProcessor,
  ImageWithAdjustment,
  FirebaseService,
  CloudinaryService,
} from '../../shared';
import {Percentages} from './Percentages';

export class ImageView extends Component {
  imageProcessor;
  originalImage;
  cloudinaryService;

  constructor(props) {
    super(props);
    this.state = {
      showOriginal: false,
      ...this.props.route.params,
    };
    this.originalImage = this.state.originalImage.clone();
    this.imageProcessor = ImageProcessor.getInstance();
    this.cloudinaryService = CloudinaryService.getInstance();
    this.firebaseService = FirebaseService.getInstance();
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.switchContainer}>
          <SwitchSelector
            style={styles.switch}
            options={options}
            onPress={value =>
              this.setState({
                showOriginal: value,
              })
            }
            {...switchConfig}
          />
        </View>
        <ImageWithAdjustment
          onImageAdjusted={this.updateOriginalImage}
          imageToEdit={this.originalImage}
          shouldRotate={this.props.route.params.shouldRotate}
          imageToShow={this.getImage()}
        />
        <Percentages
          percentageGreen={this.state.percentageGreen}
          percentageYellow={this.state.percentageYellow}
          percentageNaked={this.state.percentageNaked}
        />
      </View>
    );
  }

  updateOriginalImage = async newUri => {
    let nativeReponse = await this.imageProcessor.processImage(
      'file://' + newUri,
    );

    this.setState(async prevState => {
      let newState = {
        ...prevState,
      };
      newState.originalImage.setUri('file://' + newUri);
      newState.processedImage.setData(nativeReponse.img);
      newState.percentageGreen = nativeReponse.percentageGreen;
      newState.percentageYellow = nativeReponse.percentageYellow;
      newState.percentageNaked = nativeReponse.percentageNaked;

      // // const uri = await this.cloudinaryService.uploadPhoto(
      // //   `data:image/png;base64,${newState.processedImage.getData()}`,
      // // );
      // const uri = await this.cloudinaryService.uploadPhoto(
      //   newState.originalImage.getUri(),
      // );
      // if (uri != null) {
      //   this.firebaseService.uploadPhoto(uri);
      // }
      return newState;
    });
  };

  getImage() {
    let sourceOfImage = this.state.showOriginal
      ? this.state.originalImage
      : this.state.processedImage;
    return sourceOfImage;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
