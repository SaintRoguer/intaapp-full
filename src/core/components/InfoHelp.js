//@ts-check
import React, {useState, useEffect} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {HeaderBackButton} from '@react-navigation/stack';

function InfoHelp({navigation}) {
  useEffect(() => {
    navigation.setOptions({
      headerLeft: () => (
        <HeaderBackButton
          tintColor="white"
          //   onPress={() => {
          //     navigation.navigate('Main', {
          //       showInfo: false,
          //     });
          //   }}
        />
      ),
    });
  }, []);

  return (
    <View style={styles.container}>
      <Text>INfo</Text>
    </View>
  );
}
export default InfoHelp;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
