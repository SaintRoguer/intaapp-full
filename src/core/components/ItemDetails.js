//@ts-check
import React, {useState, useEffect} from 'react';
import {Text, StyleSheet, View} from 'react-native';
import {
  withAlertService,
  withFirebase,
  Tabs,
  uniqueId,
  GridWithNewButton,
  ImagesTaken,
  BottomRightButton,
  Info,
  withImageHandler,
} from '../../shared/';

function ItemDetails({
  firebaseService: firebase,
  alertService: alerts,
  navigation,
  imageHandler,
}) {
  const [images, setImages] = useState([]);
  const [pasturas, setPasturas] = useState([]);
  const [itemDetail, setItemDetail] = useState();
  const {item} = navigation.state.params;
  useEffect(() => {
    firebase
      .getDataFromInnerId('lotesDetails')(item.id)
      .then(item => {
        setItemDetail(item);
        setImages(item.images);
        setPasturas(item.pasturas);
      });
  }, [item.id]);
  const handleImage = imageHandler.pickImage({
    collectionName: 'lotesDetails',
    itemId: item.id,
  });
  const noop = () => {};
  return (
    <View style={styles.detailsContainer}>
      <Info item={itemDetail} />
      <Tabs
        firstTitle="Pasturas"
        secondTitle="Imagenes"
        FirstScreen={() => (
          <GridWithNewButton
            title=""
            data={pasturas}
            onEntryClick={noop}
            onNewClick={noop}
            onDeleteEntry={noop}
          />
        )}
        SecondScreen={() => <ImagesTaken images={images} loteId={item.id} />}
      />
      <BottomRightButton
        withBackground={true}
        buttons={[
          {
            name: 'upload',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Gallery');
              navigation.navigate('Imagen', imageResponse);
            },
          },
          {
            name: 'camera-retro',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Camera');
              navigation.navigate('Imagen', imageResponse);
            },
          },
        ]}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  detailsContainer: {
    height: '100%',
  },
  description: {
    fontSize: 21,
    textAlign: 'center',
  },
});

export default withImageHandler(withAlertService(withFirebase(ItemDetails)));
