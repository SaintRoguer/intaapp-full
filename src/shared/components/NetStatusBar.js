import React from 'react';
import {Text} from 'react-native';
import {warnThemeColor} from '../../configuration/colors';

export function NetStatusBar() {
  return (
    <Text
      style={{
        backgroundColor: warnThemeColor(0.9),
        color: 'white',
        fontWeight: 'bold',
        textAlign: 'center',
      }}>
      Trabajando sin conexi√≥n
    </Text>
  );
}
