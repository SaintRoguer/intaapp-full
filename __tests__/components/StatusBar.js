/**
 * @format
 */
import 'react-native';
import React from 'react';
import {NetStatusBar} from '../../src/shared/components/NetStatusBar';

import renderer from 'react-test-renderer';

describe('NetStatus component', () => {
  test('Renders correctly', () => {
    const component = renderer.create(<NetStatusBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
