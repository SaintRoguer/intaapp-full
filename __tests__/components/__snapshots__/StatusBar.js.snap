// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetStatus component Renders correctly 1`] = `
<Text
  style={
    Object {
      "backgroundColor": "rgba(237,29,29, 0.9)",
      "color": "white",
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
>
  Trabajando sin conexion
</Text>
`;
